source/ceylon/net/http/server/MatcherRule.ceylon

```diff
@@ -16,6 +17,22 @@ shared abstract class Matcher() {
             => Or(this, other);
 }
 
+"Matcher to leverage Undertow's template mechanism for path templates and path parameters.
+ It should be given to an Endpoint without combining it with other matchers.
+ 
```

Comment:
Well, another solution would be going the opposite way: leveraging as much of Undertow as possible to omit duplication of work. Maybe even leaving the path matching in Ceylon behind and replace it by using only Undertow's matching and use an instance of an Undertow handler for each ORed component. The And matcher doesn't fit there, but I don't see where it is useful anyway.

This is a rather destructive approach and if the long term goal is (as it probably should) to implement everything in Ceylon, then it's not the way to go. But from a practical point of view, if we want a rather effortless, clean and easy to maintainable implementation, I think it's definitely worth considering.

As I cannot afford more than a few hours/week working on it, I cannot do the implementation of any of this, I'm afraid. Can we in the meantime use the quick implementation I have provided so far?